job.go

```diff
@@ -69,7 +68,7 @@ type JobRow struct {
 
 	// Errors is a set of errors that occurred when the job was worked, one for
 	// each attempt. Ordered from earliest error to the latest error.
-	Errors []AttemptError
+	Errors [][]byte
```

Comment:
This seems like a fairly big downside and it's the reason I think #36 is the only viable option. I know we'll also be making use of metadata in the future and I think it's important that we keep these both as usable as possible.

Reply:
Yeah it's definitely a sizable downside.

I played around with this a little more and one thing we could do there is rename the existing `JobRow` to `JobRowRaw`:

```go
type JobRowRaw struct {
	// ID of the job. Generated as part of a Postgres sequence and generally
	// ascending in nature, but there may be gaps in it as transactions roll
	// back.
	ID int64

...
```

And then make `JobRow` embed a raw row with unmarshaled `Errors` that shadows the raw errors:

```go
type JobRow struct {
	*JobRowRaw

	Errors []AttemptError
}
```

It does work pretty well actually, but ugh, the downside is that it unfortunately does still complicate the inheritance a little much because we now have `JobRowRaw` -> `JobRow` -> `Job` which is starting to get gnarly.